vividus-plugin-azure-data-factory/src/main/java/org/vividus/azure/datafactory/steps/DataFactorySteps.java

```diff
@@ -96,10 +115,118 @@ public void runPipeline(String pipelineName, String factoryName, String resource
+            String resourceGroupName, Set<VariableScope> scopes, String variableName) throws IOException
+    {
+        RunFilterParameters filterParameters = new RunFilterParameters()
+                .withFilters(
+                        List.of(
+                                new RunQueryFilter()
+                                        .withOperand(RunQueryFilterOperand.PIPELINE_NAME)
+                                        .withOperator(RunQueryFilterOperator.EQUALS)
+                                        .withValues(List.of(pipelineName))
+                        )
+                );
+        filters.forEach(filter -> filter.getFilterType().setFilters(filterParameters, filter.getFilterValue()));
+        LOGGER.atInfo().addArgument(() -> {
+            try
+            {
+                return innersJacksonAdapter.serializeToJson(filterParameters);
+            }
+            catch (IOException e)
+            {
+                return "<unable to log filters: " + e.getMessage() + ">";
+            }
+        }).log("Collecting pipeline runs filtered by: {}");
+        List<PipelineRunInner> runs = dataFactoryManager.pipelineRuns().queryByFactory(resourceGroupName, factoryName,
+                filterParameters).innerModel().value();
+        bddVariableContext.putVariable(scopes, variableName, innersJacksonAdapter.serializeToJson(runs));
+    }
+
     private PipelineRun getPipelineRun(String resourceGroupName, String factoryName, String runId)
     {
         PipelineRun pipelineRun = dataFactoryManager.pipelineRuns().get(resourceGroupName, factoryName, runId);
         LOGGER.atInfo().addArgument(pipelineRun::status).log("The current pipeline run status is \"{}\"");
         return pipelineRun;
     }
+
+    @AsParameters
+    public static class RunFilter
+    {
+        private RunFilterType filterType;
+        private OffsetDateTime filterValue;
+
+        public RunFilterType getFilterType()
+        {
+            return filterType;
+        }
+
+        public void setFilterType(RunFilterType filterType)
+        {
+            this.filterType = filterType;
+        }
+
+        public OffsetDateTime getFilterValue()
+        {
+            return filterValue;
+        }
+
+        public void setFilterValue(OffsetDateTime filterValue)
+        {
+            this.filterValue = filterValue;
+        }
+    }
+
+    public enum RunFilterType
+    {
+        LAST_UPDATED_AFTER(RunFilterParameters::withLastUpdatedAfter),
+        LAST_UPDATED_BEFORE(RunFilterParameters::withLastUpdatedBefore);
+
+        private final BiConsumer<RunFilterParameters, OffsetDateTime> filterSetter;
+
+        RunFilterType(BiConsumer<RunFilterParameters, OffsetDateTime> filterSetter)
+        {
+            this.filterSetter = filterSetter;
+        }
+
+        public void setFilters(RunFilterParameters filterParameters, OffsetDateTime filterValue)
```

Comment:
Why it is `setFilters` look confusing, the method is applying a filter to the RunFilterParameters